hkmc2/shared/src/main/scala/hkmc2/semantics/Symbol.scala

```diff
@@ -195,7 +198,7 @@ sealed trait InnerSymbol extends Symbol:
   def subst(using SymbolSubst): InnerSymbol
 
 class ClassSymbol(val tree: Tree.TypeDef, val id: Tree.Ident)(using State)
-    extends MemberSymbol[ClassDef] with CtorSymbol with InnerSymbol:
+    extends ClassLikeSymbol with CtorSymbol with InnerSymbol:
```

Comment:
`ClassSymbol` is a `MemberSymbol[ClassDef]`, not a `MemberSymbol[ClassDef | ModuleDef]`. So `ClassLikeSymbol` should probably either take a type parameter or be a trait. Not sure if this difference is problematic for your use case. Let me know if it causes problems.

Reply:
It seems to cause problems if I rewrite it to `ClassSymbol[T <: Definition]`. The type system can't prove that `ClassLikeSymbol[ClassDef] <: ClassLikeSymbol[ClassDef | ModuleDefn]`. If I make the type parameter `T` covariant, I also have to make the type parameter `Defn` of `MemberSymbol` covariant, but that's not possible to `var defn: Opt[Defn]` in `MemberSymbol`.

Reply:
> The type system can't prove that `ClassLikeSymbol[ClassDef] <: ClassLikeSymbol[ClassDef | ModuleDefn]`.

Well, that's because it's not the case. Where is that needed?

Reply:
OK, when I experimented with it, I made `ClassLikeSymbol` take a type parameter `T <: Definition` and updated `ClassSymbol` and `ModuleSymbol` accordingly.

then, I tried to change
```
case Cls(cls: ClassLikeSymbol, path: Path)
```
into
```
case Cls(cls: ClassLikeSymbol[ClassDef | ModuleDef], path: Path)
```
which is why I need `ClassSymbol <: ClassLikeSymbol[ClassDef] <: ClassLikeSymbol[ClassDef | ModuleDef]` in some places.

If I instead changed it to
```
case Cls(cls: ClassLikeSymbol[ClassDef] | ClassLikeSymbol[ModuleDef], path: Path)
```
then I need to use the original workaround here https://github.com/hkust-taco/mlscript/pull/257#discussion_r1914145665 again, which defeats the whole point of this.

Reply:
I fixed it for you. It was not very hard. Note that the self-type I added mostly for documentation (it's not currently necessary).

BTW you don't need to use `override` when you're not actually overriding any implementation. Better leave it out.

Reply:
Thanks! I wasn't too familiar with self-types before. I'll keep it in mind for the future.